### DISCUSSION TITLE
What Typescript type should I use for `state.matches`?

### DISCUSSION THREAD

**Discussion by sabidhasan** [statelyai/xstate#4192] [General]
Hello,

I have a state machine that I am consuming in a custom hook in React using the `useMachine` helper. From the `state` that this hook exposes, I want to separately expose the `matches` function that allows a consumer to determine if the machine is in a certain state. I am wondering what Typescript type to use for this `matches` function.

Here is a code example:

```ts
import { myMachine } from "./myMachine";


interface UseMyMachine {
    matches: any;   // What Typescript type should I use here instead of `any`
}


function useMyMachine(): UseMyMachine {
    const [state, send] = useMachine(myMachine);
    // ...other code to interact with the machine

    return {
        matches: state.matches,
    }
}

```

Thanks!

**Comment by davidkpiano** [comment]
That would be `(stateValue: StateValue) => boolean`

You can import `StateValue` from `xstate`.

**Reply by Andarist** [reply]
It's better to use a type like this: `StateFrom<typeof myMachine>["matches"]`. The benefit of using that is that it's capable of reusing typegen information if that's available: [TS playground](https://www.typescriptlang.org/play?ssl=12&ssc=12&pln=12&pc=50#code/JYWwDg9gTgLgBAbzgYygUwIYzQWQ8gC2ADs0AaOAZRizQDEoIQ4BfOAM0eYCIAPAZxrZuAbjgB6cXBgBPMGn4AuOABYAdAGYAHGoBMAKFCRYiOAFd+ufEVKsOXONwACAoWnHp8MURKmz5SnAaenr6+sgQxIJwIDJ4hCRocAC8KJ7Y8TZoABQI+nDS-AAqcgrKCGwY-Ij5BY5OLoK04v5oAOZoxNzKMFBmaCK1BSBYhArUtIHcGNxwAD6OAEaitSxk+iwAlIOGxNhQ7PhJAKqWOHHWiTXDowRlVG4MTAA8rRDsMRcJpAB8ANrcEYwMb8bgAXUGLDC7DMxGQMGAkXMZy+WWym2Up1wqKueQKESi8D+TWwFEsxAAJmCUsirN8crFMoltmECugYGYoMRrnUgSDlCS0Go+Xd+OsCixIWECdFmKkLNimaR0YMQMLbgpstNuCzJHAnDB+ABaNC8eTwk1QRhQfRqkWa7jIHWDIA)
